test/primitives/array-element-test.js

```diff
@@ -128,6 +128,26 @@ describe('ArrayElement', function() {
         arrayElement.set(0, 'hello world');
         expect(arrayElement.get(0).toValue()).to.equal('hello world');
       });
+
+      it('sets the parent to children', function() {
+        var item = new minim.StringElement('foobar');
+        arrayElement.set(0, item);
+        expect(item.parents).to.include(arrayElement);
+      });
+
+      it('triggers swap on the parents', function(done) {
+        var arr = new minim.ArrayElement([1]);
+        arr.on('swap', done);
+        arr.set(1, 2);
+      });
+
+      context('when a value is changed directly', function() {
+        it('triggers swap on the parents', function(done) {
```

Comment:
Each event is on a separate node. So when you trigger an event on a node, it triggers the swap events all of the way up. This means you can listen for changes on specific parts of the tree.